src/main/java/io/hops/hopsworks/expat/migrations/projects/provenance/UpdateProvenance.java

```diff
@@ -0,0 +1,406 @@
+      .build();
+  }
+  
+  private void close() throws SQLException, IOException {
+    if(connection != null) {
+      connection.close();
+    }
+    if(httpClient != null) {
+      httpClient.close();
+    }
+  }
+  
+  @Override
+  public void migrate() throws MigrationException {
+    LOGGER.info("provenance migration");
+    DistributedFileSystemOps dfso = null;
+    try {
+      setup();
+      dfso = HopsClient.getDFSO(connection);
+      traverseElements(projectMigrate(dfso), datasetMigrate(dfso));
+    } catch (IllegalStateException | SQLException | ConfigurationException | GeneralSecurityException | IOException e) {
+      throw new MigrationException("error", e);
+    } finally {
+      if (dfso != null) {
+        dfso.close();
+      }
+      try {
+        close();
+      } catch (SQLException | IOException e) {
+        throw new MigrationException("error", e);
+      }
+    }
+  }
+  
+  @Override
+  public void rollback() throws RollbackException {
+    LOGGER.info("provenance rollback");
+    DistributedFileSystemOps dfso = null;
+    try {
+      setup();
+      dfso = HopsClient.getDFSO(connection);
+      traverseElements(projectRollback(dfso), datasetRollback(dfso));
+      ElasticClient.deleteAppProvenanceIndex(httpClient, elastic, elasticUser, elasticPass);
+    } catch (IllegalStateException | SQLException | ConfigurationException | GeneralSecurityException | IOException e) {
+      throw new RollbackException("error", e);
+    } finally {
+      if (dfso != null) {
+        dfso.close();
+      }
+      try {
+        close();
+      } catch (SQLException | IOException e) {
+        throw new RollbackException("error", e);
+      }
+    }
+  }
+  
+  private <E extends Exception> void traverseElements(CheckedConsumer<ProjectParams, E> projectAction,
+    CheckedConsumer<DatasetParams, E> datasetAction)
+    throws E, SQLException, ConfigurationException, IOException, GeneralSecurityException {
+    setup();
+    
+    PreparedStatement allProjectsStmt = null;
+    PreparedStatement projectInodeStmt = null;
+    PreparedStatement allProjectDatasetsStmt = null;
+    PreparedStatement datasetInodeStmt = null;
+    try {
+      connection.setAutoCommit(false);
+      //get all projects
+      allProjectsStmt = connection.prepareStatement(GET_ALL_PROJECTS);
+      ResultSet allProjectsResultSet = allProjectsStmt.executeQuery();
+      
+      while (allProjectsResultSet.next()) {
+        //get project inode
+        projectInodeStmt = getProjectInodeStmt(allProjectsResultSet);
+        ResultSet projectInodeResultSet = projectInodeStmt.executeQuery();
+        if(!projectInodeResultSet.next()) {
+          throw new IllegalStateException("project inode not found");
+        }
+        
+        ProjectParams projectParams = ProjectParams.instance(allProjectsResultSet, projectInodeResultSet);
+        LOGGER.debug("processing project:{}", projectParams.projectName);
```

Comment:
Can you set the project logs to info so we can see how it's progressing?

Reply:
fixed